core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -178,6 +182,13 @@ public void initialize(BatchJoinerRuntimeContext context) {
     init(inputSchemas, collector);
     collector.getOrThrowException();
     outputSchema = getOutputSchema(inputSchemas, collector);
+    if (outputSchema == null) {
+      // Could not derive output schema from input schema. Try to get output schema from config.
+      outputSchema = conf.getOutputSchema(collector);
+      if (outputSchema == null) {
```

Comment:
this should never be null, it should be validated in the configurePipeline() method.

Reply:
in configurePipeline(), you should validate that a valid output schema is provided if the input schemas are unknown.

Reply:
moved validation to configurePipeline()